src/Sylius/Behat/Context/Api/Admin/ManagingOrdersContext.php

```diff
@@ -0,0 +1,211 @@
+     * @Then its state should be :state
+     */
+    public function itsStateShouldBe(string $state): void
+    {
+        Assert::true(
+            $this->responseChecker->hasValue(
+                $this->client->showByIri($this->iriConverter->getIriFromItem($this->sharedStorage->get('order'))),
+                'state',
+                strtolower($state)
+            ),
+            sprintf('Order have different state then %s', $state)
+        );
+    }
+
+    /**
+     * @Then it should have shipment in state :state
+     */
+    public function itShouldHaveShipmentState(string $state): void
+    {
+        $shipmentsIri = $this->responseChecker->getValue(
+            $this->client->showByIri($this->iriConverter->getIriFromItem($this->sharedStorage->get('order'))),
+            'shipments'
+        );
+
+        Assert::true(
+            $this->responseChecker->hasValue($this->client->showByIri($shipmentsIri[0]), 'state', strtolower($state)),
+            sprintf('Shipment for this order is not %s', $state)
+        );
+    }
+
+    /**
+     * @Then it should have payment state :state
+     */
+    public function itShouldHavePaymentState($state): void
+    {
+        $paymentsIri = $this->responseChecker->getValue(
+            $this->client->showByIri($this->iriConverter->getIriFromItem($this->sharedStorage->get('order'))),
+            'payments'
+        );
+
+        Assert::true(
+            $this->responseChecker->hasValue($this->client->showByIri($paymentsIri[0]), 'state', strtolower($state)),
+            sprintf('payment for this order is not %s', $state)
+        );
+    }
+
+    /**
+     * @Then /^there should be(?:| only) (\d+) payments?$/
+     */
+    public function theOrderShouldHaveNumberOfPayments(int $number): void
+    {
+        $orderIri = $this->iriConverter->getIriFromItem($this->sharedStorage->get('order'));
+
+        Assert::count($this->responseChecker->getValue($this->client->showByIri($orderIri), 'payments'), $number);
+    }
+
+    /**
+     * @Then /^(this order) should have order payment state "([^"]+)"$/
+     */
+    public function theOrderShouldHavePaymentState(OrderInterface $order, string $paymentState): void
+    {
+        Assert::true(
+            $this->responseChecker->hasValue($this->client->show($order->getNumber()), 'paymentState', strtolower($paymentState)),
+            sprintf('Order %s does not have %s payment state', $order->getNumber(), $paymentState)
+        );
+    }
+
+    /**
+     * @Then /^I should not be able to cancel (this order)$/
+     */
+    public function iShouldNotBeAbleToCancelThisOrder(OrderInterface $order): void
+    {
+        $this->iCancelThisOrder($order);
+        Assert::contains(
+            $this->responseChecker->getError($this->client->getLastResponse()),
+            'Transition "cancel" cannot be applied on state "cancelled"'
+        );
+    }
+
+    /**
+     * @Then /^the order's total should(?:| still) be "(\-)?(?:€|£|￥|\$)((?:\d+\.)?\d+)"$/
```

Comment:
```suggestion
     * @Then /^the order's total should(?:| still) be "([^"]+)"$/
```
As in src/Sylius/Behat/Context/Ui/Admin/ManagingOrdersContext.php:317